frontend/js/modules/watchlist.js

```diff
@@ -0,0 +1,190 @@
+/**
+*  vim:set sw=2 ts=2 sts=2 ft=javascript expandtab:
+*
+*  # Bookmark List module
+*
+*  ## License
+*
+*  Licensed to the Apache Software Foundation (ASF) under one
+*  or more contributor license agreements.  See the NOTICE file
+*  distributed with this work for additional information
+*  regarding copyright ownership.  The ASF licenses this file
+*  to you under the Apache License, Version 2.0 (the
+*  "License"); you may not use this file except in compliance
+*  with the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*
+*  ## Description
+*
+*  This module regroups all bookmarks from a logged users : groups, pads and,
+*  _soon_, other users.
+*/
+
+module.exports = (function () {
+    'use strict';
+    // Global dependencies
+    var m  = require('mithril');
+    var ld = require('lodash');
+  
+    // Local dependencies
+    var conf      = require('../configuration.js');
+    var auth      = require('../auth.js');
+    var u         = auth.userInfo;
+    var layout    = require('./layout.js');
+    var groupWatch= require('./group-watch.js');
+    var padWatch  = require('./pad-watch.js');
+    var model     = require('../model/group.js');
+  
+    var watchlist = {};
+  
+    /**
+    * ## Controller
+    *
+    * Used for module state and actions.
+    */
+  
+   watchlist.controller = function () {
+      if (!auth.isAuthenticated()) {
+        conf.unauthUrl(true);
+        return m.route('/login');
+      }
+      document.title = conf.LANG.WATCHLIST.TITLE + ' - ' + conf.SERVER.title;
+  
+      var c = {};
+  
+      /**
+      * ### computeWatchlist
+      *
+      * `computeWatchlist` is an internal function that gathers watchlisted groups
+      * and pads.
+      */
+  
+      c.computeWatchlist = function () {
+        var uWatched = u().watchlist;
+        console.log(uWatched);
+        console.log(model.watchlist());
+        var items  = function (data, watched) {
+          return  ld(data)
+            .values()
+            .filter(function (v) { return ld.includes(watched, v._id); })
+            .sortBy('name')
+            .value();
+        };
+        console.log(items(model.watchlist().groups, uWatched.groups));
```

Comment:
Cache the results of model.watchlist() in a local variable, to avoid multiple function execution